src/main/java/com/scottlogic/filter_sort/FilterKeywordSortFrequency.java

```diff
@@ -0,0 +1,73 @@
+package com.scottlogic.filter_sort;
+
+import com.scottlogic.SortOrder;
+import com.scottlogic.UserPost;
+import com.scottlogic.filter.Filter;
+import com.scottlogic.filter.FilterContentKeyword;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NavigableMap;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FilterKeywordSortFrequency implements Filter {
+
+    private String keyword;
+    private SortOrder sortOrder;
+
+    public FilterKeywordSortFrequency(String filterWord, SortOrder orderToSort) {
+        keyword = filterWord;
+        sortOrder = orderToSort;
+    }
+
+    @Override
+    public List<UserPost> filter(List<UserPost> inputList) {
```

Comment:
This is awesome, perfect use of maps. 

This method is quite long. Could we isolate the populating of the Map and the sorting into smaller readable methods?

Reply:
Placed them into their own methods as you described - let me know what you think